packages/flutter/lib/src/foundation/stack_frame.dart

```diff
@@ -0,0 +1,209 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' show hashValues;
+
+import 'package:meta/meta.dart';
+
+/// A object representation of a frame from a stack trace.
+///
+/// {@tool sample}
+///
+/// For example, a caller that wishes to traverse the stack could use
+///
+/// ```dart
+/// final List<StackFrame> currentFrames = StackFrame.fromStackTrace(StackTrace.current);
+/// ```
+///
+/// To create a traversable parsed stack.
+/// {@end-tool}
+@immutable
+class StackFrame {
+  /// Creates a new StackFrame instance.
+  ///
+  /// All parameters must not be null. The [className] may be the empty string
+  /// if there is no class (e.g. for a top level library method).
+  const StackFrame(
+    this.number, {
+    @required this.column,
+    @required this.line,
+    @required this.packageScheme,
+    @required this.package,
+    @required this.packagePath,
+    this.className = '',
+    @required this.method,
+    this.isConstructor = false,
+  })  : assert(number != null),
+        assert(column != null),
+        assert(line != null),
+        assert(method != null),
+        assert(packageScheme != null),
+        assert(package != null),
+        assert(packagePath != null),
+        assert(className != null),
+        assert(isConstructor != null);
+
+  /// A stack frame representing an asynchronous suspension.
+  static const StackFrame asynchronousSuspension = StackFrame(
+    -1,
```

Comment:
Should this also be a named parameter? (currently, I'd have to click through to StackFrame to learn what the -1 means.)

Reply:
I had it as named but it felt awkward.  I could make it named again.

Reply:
you could also just use nulls, fwiw.